<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>840</width>
    <height>520</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>840</width>
    <height>520</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1024</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <item>
    <widget class="GLWidget" name="openGLW">
     <property name="minimumSize">
      <size>
       <width>540</width>
       <height>500</height>
      </size>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Музика:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="playOrStopBtn">
           <property name="text">
            <string>►</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Швидкість анімації</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="speedSlider">
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="pageStep">
          <number>2</number>
         </property>
         <property name="value">
          <number>2</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="comeIn">
           <property name="font">
            <font>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Надійшло: 0</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="comeOut">
           <property name="font">
            <font>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Відремонтовано: 0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableWidget" name="tableWidget">
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAsNeeded</enum>
         </property>
         <property name="sizeAdjustPolicy">
          <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
         </property>
         <property name="rowCount">
          <number>2</number>
         </property>
         <property name="columnCount">
          <number>4</number>
         </property>
         <attribute name="horizontalHeaderDefaultSectionSize">
          <number>65</number>
         </attribute>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <row/>
         <row/>
         <column>
          <property name="text">
           <string notr="true">№</string>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Кількість</string>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Сер. час, 
хв</string>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Відступ, 
хв</string>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
         </column>
         <item row="0" column="0">
          <property name="text">
           <string>1</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsDropEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="1">
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item row="0" column="2">
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item row="0" column="3">
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item row="1" column="0">
          <property name="text">
           <string>2</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsDropEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item row="1" column="1">
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item row="1" column="2">
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item row="1" column="3">
          <property name="text">
           <string>1</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="addRowBtn">
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Додати рядок</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="removeRowBtn">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Видалити рядок</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="trainsCountLabel">
           <property name="text">
            <string>Кількість вагонів:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="trainsCount"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Почати моделювання</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>glwidget.h</header>
   <slots>
    <slot>animate()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>818</x>
     <y>508</y>
    </hint>
    <hint type="destinationlabel">
     <x>693</x>
     <y>416</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speedSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Widget</receiver>
   <slot>speedChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>818</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>701</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addRowBtn</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>addRowToTable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>629</x>
     <y>388</y>
    </hint>
    <hint type="destinationlabel">
     <x>701</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeRowBtn</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>removeRowFromTable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>761</x>
     <y>408</y>
    </hint>
    <hint type="destinationlabel">
     <x>824</x>
     <y>438</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playOrStopBtn</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>switchMusicStatement()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>763</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>670</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>run()</slot>
  <slot>speedChanged(int)</slot>
  <slot>addRowToTable()</slot>
  <slot>removeRowFromTable()</slot>
  <slot>switchMusicStatement()</slot>
 </slots>
</ui>
